pkg/clusterversion/clusterversion.go

```diff
@@ -90,3 +99,268 @@ func (builder *Builder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WithDesiredUpdateImage adds the desired image to the clusterversion struct.
+func (builder *Builder) WithDesiredUpdateImage(desiredUpdateImage string, force bool) *Builder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
+
+		return builder
+	}
+
+	glog.V(100).Info("Adding the desired image %s to clusterversion %s",
+		desiredUpdateImage, builder.Definition.Name)
+
+	if desiredUpdateImage == "" {
+		glog.V(100).Infof("The desiredUpdateImage is empty")
+
+		builder.errorMsg = "'desiredUpdateImage' cannot be empty"
+
+		return builder
+	}
+
+	builder.Definition.Spec.DesiredUpdate = &v1.Update{Image: desiredUpdateImage, Force: force}
+
+	return builder
+}
+
+// WithDesiredUpdateChannel adds the desired channel to the clusterversion struct.
+func (builder *Builder) WithDesiredUpdateChannel(updateChannel string) *Builder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
```

Comment:
Can be removed once https://github.com/openshift-kni/eco-goinfra/pull/475/files#r1644903566 is added

Reply:
removed, thanks